compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Utils/AMDAIEDmaUtils.cpp

```diff
@@ -292,6 +304,60 @@ LogicalResult combineAccessPatterns(
   return success();
 }
 
+LogicalResult expandLargeDimIntoLinearDims(
+    MLIRContext *ctx, const SmallVector<OpFoldResult> &offsets,
+    const SmallVector<OpFoldResult> &sizes,
+    const SmallVector<OpFoldResult> &strides,
+    SmallVector<OpFoldResult> &newOffsets, SmallVector<OpFoldResult> &newSizes,
+    SmallVector<OpFoldResult> &newStrides, ArrayRef<int64_t> maxSizes) {
+  assert(offsets.size() == sizes.size() && offsets.size() == strides.size() &&
+         "expected same number of offsets, sizes and strides");
+  assert(maxSizes.size() >= sizes.size() &&
+         "expected `maxSizes` to have at least as many elements as `sizes`");
+  bool expandableLinearDimsFound = false;
+  if (offsets.size() == 0) return success(expandableLinearDimsFound);
```

Comment:
why not directly return success()?

Reply:
Because no expansion took place, so that the user can avoid/abort a rewrite if not needed. This function is written in the same way as `foldLinearDims` and it's documented, but we could reconsider how we return that info:

> /// Expand dimensions within a strided access pattern that exceed a maximum
/// size. Returns `success` if expansion took place. The `maxSizes` array is
/// expected to be provided in the same order as `sizes` and they are compared
/// from right to left (innermost to outermost).